app/src/main/java/com/codingblocks/cbonlineapp/adapters/CourseDataAdapter.kt

```diff
@@ -48,51 +48,42 @@ class CourseDataAdapter(
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
 
-        var viewHolder: RecyclerView.ViewHolder? = null
+        var viewHolder: RecyclerView.ViewHolder = MyCoursesViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.my_course_card_horizontal, parent, false))
```

Comment:
Why . are you setting viewHolder even before knowing the type

Reply:
when you know there are only two conditions, set the initial value as one and change it if it matches the other.

Reply:
it is better than null asserting the viewholder which is a crucial component

Reply:
I've improved the condition, let me know if you agree